File: sdklibrary/src/main/java/com/loner/android/sdk/webservice/network/apis/BaseRequest.java
package com.loner.android.sdk.webservice.network.apis;

import com.loner.android.sdk.webservice.interfaces.APIEndpoints;
import com.loner.android.sdk.webservice.network.networking.RetrofitClientInstance;
import com.loner.android.sdk.webservice.network.networking.RequestConfig;

/**
 * This files used as base class for other API classes.
 */
public abstract class BaseRequest implements RequestConfig {
    public APIEndpoints interfaceAPI = RetrofitClientInstance.getRetrofitInstance().create(APIEndpoints.class);
}

File: LonerSdkLibrary/src/main/java/com/loner/android/sdk/webservice/network/apis/BaseRequest.kt
package com.loner.android.sdk.webservice.network.apis

import com.loner.android.sdk.webservice.interfaces.APIEndpoints
import com.loner.android.sdk.webservice.network.networking.RequestConfig
import com.loner.android.sdk.webservice.network.networking.RetrofitClientInstance

/**
 * This files used as base class for other API classes.
 */
abstract class BaseRequest : RequestConfig {
    var interfaceAPI = RetrofitClientInstance.retrofitInstance!!.create(APIEndpoints::class.java)
}